examples/tutorials/testing.md

```diff
@@ -0,0 +1,729 @@
+    Error,
+    "Failed to fetch user: 404",
+  );
+});
+```
+
+## Mocking in tests
+
+Mocking is an essential technique for isolating the code being tested from its
+dependencies. Deno provides built-in utilities and third-party libraries for
+creating mocks.
+
+### Basic Mocking
+
+For simple cases, you can override global or imported objects directly:
+
+```ts
+// database.ts
+export async function getUserFromDB(id: string) {
+  // In reality, this would connect to a database
+  // For this example, we'll just simulate an async operation
+  return { id, name: "User " + id };
+}
+
+// service.ts
+import { getUserFromDB } from "./database.ts";
+
+export async function getUserDetails(id: string) {
+  const user = await getUserFromDB(id);
+  return {
+    ...user,
+    formattedName: user.name.toUpperCase(),
+    timestamp: new Date().toISOString(),
+  };
+}
+
+// service_test.ts
+import { assertEquals } from "jsr:@std/assert";
+import { getUserDetails } from "./service.ts";
+import * as db from "./database.ts";
+
+Deno.test("getUserDetails formats the name correctly", async () => {
+  // Replace the real implementation with a mock
+  const originalGetUserFromDB = db.getUserFromDB;
+
+  // Create a mock that returns a predictable result
+  db.getUserFromDB = async (id: string) => {
+    return { id, name: "Test User" };
+  };
+
+  try {
+    // Mock the Date object to return a fixed timestamp
+    const originalDate = globalThis.Date;
+    const fixedDate = new Date("2023-01-01T12:00:00Z");
+    globalThis.Date = class extends originalDate {
+      constructor() {
+        super();
+        return fixedDate;
+      }
+    } as DateConstructor;
+
+    const result = await getUserDetails("123");
+
+    assertEquals(result, {
+      id: "123",
+      name: "Test User",
+      formattedName: "TEST USER",
+      timestamp: "2023-01-01T12:00:00.000Z",
+    });
+  } finally {
+    // Restore the original implementations
+    db.getUserFromDB = originalGetUserFromDB;
+    globalThis.Date = originalDate;
+  }
+});
+```
+
+### Using Spy Functions
+
+Spies allow you to track function calls without changing their behavior:
+
+```ts
+import { spy } from "jsr:@std/testing/mock";
+
+Deno.test("spy example", () => {
+  // Create a spy on console.log
+  const consoleSpy = spy(console, "log");
+
+  // Call the function we're spying on
+  console.log("Hello");
+  console.log("World");
+
+  // Verify the function was called correctly
+  assertEquals(consoleSpy.calls.length, 2);
+  assertEquals(consoleSpy.calls[0].args, ["Hello"]);
+  assertEquals(consoleSpy.calls[1].args, ["World"]);
+
+  // Restore the original function
+  consoleSpy.restore();
+});
+```
+
+For more advanced mocking techniques, check our
+[dedicated guide on mocking in Deno](/examples/mocking_tutorial/).
+
+## Test parallelization
+
+By default, Deno runs tests sequentially for predictable results. However, you
+can run tests in parallel for faster execution:
```

Comment:
Not sure what this means, there is no way to run tests itself in parallel. There is only the `--parallel` flag to run test files/suites in parallel.

Reply:
Have removed, my misunderstanding